doc/source/performance/xtradb_performance_improvements_for_io-bound_highly-concurrent_workloads.rst

```diff
@@ -29,6 +29,31 @@ Even the above implementation does not fully resolve the buffer pool free list m
 
 When ``legacy`` option is set, server will used the upstream algorithm and when the ``backoff`` is selected, |Percona| implementation will be used.
 
+.. _lru_manager_threads:
+
+Multi-threaded LRU manager threads
+==================================
+
+|Percona Server| :rn:`5.7.10-3` has introduced LRU manager threads out of the |InnoDB| page cleaner thread, that perform LRU flushes and evictions to refill the buffer pool free lists. Before this implementation this was done by the page_cleaner thread together with the flush list flushes.
```

Comment:
Percona Server has introduced a true multithreaded LRU flushing. In this scheme, each buffer pool instance has its own dedicated LRU manager thread that is tasked with performing LRU flushes and evictions to refill the free list of that buffer pool instance. Existing multithreaded flusher no longer does any LRU flushing and is tasked with flush list flushing only.

This has been done to address the shortcomings of the existing MySQL 5.7 multithreaded flusher:
* All threads still synchronize on each coordinator thread iteration. If a particular flushing job is stuck on one of the worker threads, the rest will idle until the stuck one completes.
* The coordinator thread heuristics focus on flush list adaptive flushing without considering the state of free lists, which might be in need of urgent refill for a subset of buffer pool instances on a loaded server.
* LRU flushing is serialized with flush list flushing for each buffer pool instance.